homeassistant/components/hvv_departures/config_flow.py

```diff
@@ -0,0 +1,224 @@
+                )
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+
+        return self.async_show_form(
+            step_id="user", data_schema=SCHEMA_STEP_USER, errors=errors
+        )
+
+    async def async_step_station(self, user_input=None):
+        """Handle the step where the user inputs his/her station."""
+        errors = {}
+        if user_input is not None:
+
+            check_name = await self.hub.gti.checkName(
+                {"theName": {"name": user_input["station"]}, "maxList": 20}
+            )
+
+            results = check_name.get("results")
+
+            if not results:
+                errors["base"] = "no_results"
+
+                return self.async_show_form(
+                    step_id="station", data_schema=SCHEMA_STEP_STATION, errors=errors
+                )
+
+            self.stations = {
+                "{} ({})".format(x.get("name"), x.get("type")): x for x in results
+            }
+
+            schema = vol.Schema(
+                {vol.Required("station"): vol.In(list(self.stations.keys()))}
+            )
+
+            return self.async_show_form(
+                step_id="station_select", data_schema=schema, errors=errors
+            )
+
+        return self.async_show_form(
+            step_id="station", data_schema=SCHEMA_STEP_STATION, errors=errors
+        )
+
+    async def async_step_station_select(self, user_input=None):
+        """Handle the step where the user inputs his/her station."""
+        errors = {}
+        if user_input is not None:
+            self.data.update({"station": self.stations[user_input["station"]]})
+
+            # get station information
+            station_information = await self.hub.gti.stationInformation(
+                {"station": self.data["station"]}
+            )
+
+            self.data.update({"stationInformation": station_information})
+
+            title = self.data["station"]["name"]
+
+            return self.async_create_entry(title=title, data=self.data)
+
+        return self.async_show_form(
+            step_id="station_select", data_schema=SCHEMA_STEP_STATION, errors=errors
+        )
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get options flow."""
+        return OptionsFlowHandler(config_entry)
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Options flow handler."""
+
+    def __init__(self, config_entry):
+        """Initialize HVV Departures options flow."""
+        self.config_entry = config_entry
+        self.options = dict(config_entry.options)
+        self.filters = {}
+        self.hub = None
+
+    async def async_step_init(self, user_input=None):
+        """Manage the options."""
+        errors = {}
+        if self.filters == {}:
+
+            try:
```

Comment:
Only wrap the line where an error could occur.

Reply:
Why is this a requirement? It makes it super complicated, as now I get "UnboundLocalError: local variable 'departure_list' referenced before assignment" errors and I have to check that before. Other integrations are also executing more than the falsy line inside the try.

```python
            try:
                departure_list = await self.hub.gti.departureList(
                    {
                        "station": self.config_entry.data["station"],
                        "time": {"date": "heute", "time": "jetzt"},
                        "maxList": 5,
                        "maxTimeOffset": 200,
                        "useRealtime": True,
                        "returnFilters": True,
                    }
                )
            except CannotConnect:
                errors["base"] = "cannot_connect"
            except InvalidAuth:
                errors["base"] = "invalid_auth"

            if departure_list:
                self.filters = {
                f"{x['serviceName']}, {x['label']}": x
                    for x in departure_list.get("filter")
                }
```

Reply:
Wrapping only the falsy line inside the try makes the code easier to read and maintain because you can directly see where an error could occur. I would recommend to declare the variable beforehand and then use it inside of the try.
Furthermore, I would refer back to the [documentation](https://developers.home-assistant.io/docs/en/next/creating_component_code_review.html) about other integrations not following this rule:

> Not all existing code follow the requirements in this checklist. This cannot be used as a reason to not follow them!

Moreover, multiple lines can be inside the try if the are falsy and return the same/similar exception.

Reply:
I also removed the authenticate call, as I noticed that everything is checked directly after that by departureList.

Reply:
I also removed the authenticate, because everything is checked by departureList directly after that